OSDOCS-1770 Installing a cluster on GCP in a restricted network

codyhoag opened the pull request:
https://issues.redhat.com/browse/OSDOCS-1770

Previews:
* [Installing a cluster on GCP in a restricted network](https://deploy-preview-32482--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.html)
* [Support matrix update](https://deploy-preview-32482--osdocs.netlify.app/openshift-enterprise/latest/installing/installing-preparing.html#supported-installation-methods-for-different-platforms)

codyhoag commented:
@staebler PTAL. I followed a similar path as what was recommended for AWS. Thanks!

staebler reviewed (commented):
LGTM

codyhoag commented:
@tsze-redhat can you confirm from a QE perspective? Thanks!

tsze-redhat commented:
Is this for adding a page for IPI Restricted cluster?

Assuming it is,
1 why do we need an existing vpc?

2 Item 12 in 
https://deploy-preview-32482--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.html#installation-gcp-config-yaml_installing-restricted-networks-gcp-installer-provisioned

under
additionalTrustBundle: | 
    -----BEGIN CERTIFICATE-----
    <MY_TRUSTED_CA_CERT>
    -----END CERTIFICATE-----
    
texts need to have two spaces from beginning.

Right now, there appears to be 4.

Can you confirm?

3 Hyperthreading for compute under
compute: 
- hyperthreading: Enabled

is marked as remark #3 only but it should be #3 & #4 (similar to control).

4 Configuring the cluster-wide proxy during installation should be optional
https://deploy-preview-32482--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.html#installation-configure-proxy_installing-restricted-networks-gcp-installer-provisioned

In fact, why do we have this section?
It is not there in Restricted network UPI page.

This section only appears in "Installing a cluster on GCP into an existing VPC" page.

Please feel free to ping me if you need any clarification on any of the above.

Thanks.

codyhoag commented:
@tsze-redhat a few answers below:
> Is this for adding a page for IPI Restricted cluster?

Yes

> 1. why do we need an existing vpc?

Matthew explained it well [here](https://github.com/openshift/openshift-docs/pull/31935#discussion_r628534333). A snippet from his explanation to me: `The point of needing to BYO VPC is that either (1) the mirror registry needs to be in the VPC or (2) the VPC needs to have firewall rules or peering to access the mirror registry hosted elsewhere. In either case, the VPC needs to exist before installation.`

> 2. texts need to have two spaces from beginning. Right now, there appears to be 4.

Fixed

> 3. Hyperthreading for compute is marked as remark #3 only but it should be #3 & #4 (similar to control).

Fixed
> 4. Configuring the cluster-wide proxy during installation should be optional. In fact, why do we have this section? It is not there in Restricted network UPI page.

I've been putting this in the restricted assemblies based on feedback I receieved during the vSphere review process from Jinyun [here](https://github.com/openshift/openshift-docs/pull/31268#issuecomment-814536752). From what I understand, this used to not be supported for IPI installs, but is now supported as of recently. I'm not sure why it is not there for GCP UPI, but I can add it there in a separate PR if that would be relevant there.

**Let me know if there's any other feedback or any changes for this.**

tsze-redhat commented:
Given that user must provide vpc, 
6, 7, 8 in https://deploy-preview-32482--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.html#installation-gcp-config-yaml_installing-restricted-networks-gcp-installer-provisioned
should not begin with "If you use an existing VPC, " 

The rest is good.

Thanks.

maxwelldb commented on installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.adoc:
```diff
@@ -0,0 +1,62 @@
+
+[IMPORTANT]
+====
+While you can install an {product-title} cluster by using mirrored installation release content, your cluster still requires internet access to use the GCP APIs.
```
```suggestion
You can install an {product-title} cluster by using mirrored installation release content, but your cluster will require internet access to use the GCP APIs.
```

maxwelldb commented on installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.adoc:
```diff
@@ -0,0 +1,62 @@
+====
+Because the installation media is on the bastion host, use that computer to complete all installation steps.
+====
+* You have an existing VPC in GCP. When installing to a restricted network using installer-provisioned infrastructure, you cannot use the installer-provisioned VPC. You must use a user-provisioned VPC that satisfies one of the following requirements:
```
```suggestion
* You have an existing VPC in GCP. While installing a cluster in a restricted network that uses installer-provisioned infrastructure, you cannot use the installer-provisioned VPC. You must use a user-provisioned VPC that satisfies one of the following requirements:
```

Not sure on "While," specifically. Could also be "If."

maxwelldb commented on installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.adoc:
```diff
@@ -0,0 +1,62 @@
+Because the installation media is on the bastion host, use that computer to complete all installation steps.
+====
+* You have an existing VPC in GCP. When installing to a restricted network using installer-provisioned infrastructure, you cannot use the installer-provisioned VPC. You must use a user-provisioned VPC that satisfies one of the following requirements:
+** Contains the mirror registry.
```
ISG: Don't punctuate the ends of fragments.

Introductory sentence into the list reads a little awk to me. Not major, but maybe consider rephrasing.

maxwelldb commented on installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.adoc:
```diff
@@ -0,0 +1,62 @@
+** Contains the mirror registry.
+** Has firewall rules or a peering connection to access the mirror registry hosted elsewhere.
+* You reviewed details about the xref:../../architecture/architecture-installation.adoc#architecture-installation[{product-title} installation and update] processes.
+* If you use a firewall, you must xref:../../installing/install_config/configuring-firewall.adoc#configuring-firewall[configure it to allow the sites] that your cluster requires access to. While you might need to grant access to more sites, you must grant access to `*.googleapis.com` and `accounts.google.com`.
```
Consider rephrasing or splitting.

maxwelldb commented on installing/installing_gcp/installing-restricted-networks-gcp-installer-provisioned.adoc:
```diff
@@ -0,0 +1,62 @@
+** Has firewall rules or a peering connection to access the mirror registry hosted elsewhere.
+* You reviewed details about the xref:../../architecture/architecture-installation.adoc#architecture-installation[{product-title} installation and update] processes.
+* If you use a firewall, you must xref:../../installing/install_config/configuring-firewall.adoc#configuring-firewall[configure it to allow the sites] that your cluster requires access to. While you might need to grant access to more sites, you must grant access to `*.googleapis.com` and `accounts.google.com`.
+* If you do not allow the system to manage identity and access management (IAM), then a cluster administrator can xref:../../installing/installing_gcp/manually-creating-iam-gcp.adoc#manually-creating-iam-gcp[manually create and maintain IAM credentials]. Manual mode can also be used in environments where the cloud IAM APIs are not reachable.
```
Must the cluster admin? Not sure if this is a prereq or an option.

codyhoag replied:
Yeah, I think this is a tweener that gives suggestions for a particular scenario. Since this is reproduced throughout the AWS/Azure/GCP install assemblies, I'll leave it for now and follow up with a possible stronger statement.

maxwelldb reviewed (commented):
Looks copacetic. 

I made a few comments/suggestions, but nothing major. 

The list punctuation thing should be resolved prior to merging.

codyhoag commented:
/cherrypick enterprise-4.8

codyhoag commented:
/cherrypick enterprise-4.7

codyhoag commented:
/cherrypick enterprise-4.6

codyhoag commented:
/cherrypick enterprise-4.5
